tools/mpremote/mpremote/mip.py

```diff
@@ -118,6 +123,8 @@ def _install_json(transport, package_json_url, index, target, version, mpy):
         _download_file(transport, file_url, fs_target_path)
     for target_path, url in package_json.get("urls", ()):
         fs_target_path = target + "/" + target_path
+        if url.split(":", 1)[0] not in ("http", "https", "github"):
```

Comment:
Does this need to include "gitlab" now as well?

Also, is it simpler and equivalent to just write `if not url.startswith(("http", "https", "github", "gitlab")): ...`?

Reply:
Actually, I should have suggested `if not url.startswith(("http://", "https://", "github:", "gitlab:")): ...`.

Reply:
Done. I have also consolidated the prefixes into a module variable as for #12476. These two PRs will now have a conflict when merged.

This could be resolved by:
- patching the second after merging the first or
- merging the PRs or
- changing to stacked PRs.